docs/quickstarts/astro.mdx

```diff
@@ -0,0 +1,211 @@
+
+Inside your `src/` directory, update the `.env.d.ts` file to ensure that the locals added by the middleware are properly typed.
+
+```ts filename="src/.env.d.ts" {3}
+/// <reference types="astro/client" />
+/// <reference types="@clerk/astro/env" />
+```
+
+### Update `astro.config.mjs`
+
+To configure Clerk in your astro application, you will need to update your `astro.config.mjs`.
+
+- Add the Clerk integration to the `integrations` list.
+- Install an [SSR adapter](https://docs.astro.build/en/guides/server-side-rendering/#official-adapters). This is required for the integration to work. For this quickstart we chose the [`@astrojs/node`](https://docs.astro.build/en/guides/integrations-guide/node/) adapter.
+- Set `output` to `server`. This is required when deploying to a host supporting SSR.
+
+```ts filename="astro.config.mjs" {2,3,6-8}
+import { defineConfig } from "astro/config";
+import node from "@astrojs/node";
+import clerk from "@clerk/astro";
+
+export default defineConfig({
+  integrations: [clerk()],
+  adapter: node({ mode: "standalone" }),
+  output: "server",
+});
+```
+
+### Add middleware to your application
+
+[`clerkMiddleware()`](/docs/references/astro/clerk-middleware) grants you access to user authentication state throughout your application, on any route or page. It also allows you to protect specific routes from unauthenticated users. To add `clerkMiddleware()` to your application, follow these steps:
+
+1. Create a `middleware.ts` file inside your `src/` directory.
+2. In your `middleware.ts` file, export an `onRequest` constant and assign the result of the `clerkMiddleware` function to it.
+
+```tsx filename="src/middleware.ts"
+import { clerkMiddleware } from "@clerk/astro/server";
+
+export const onRequest = clerkMiddleware();
+```
+
+3. By default, `clerkMiddleware()` will not protect any routes. All routes are public and you must opt-in to protection for routes. See the [`clerkMiddleware()` reference](/docs/references/astro/clerk-middleware) to learn how to require authentication for specific routes.
+
+### Add components to your app
+
+Clerk offers components that allow you to protect your pages. These components are used to control the visibility of your pages based on the user's authentication state.
+
+
+```astro filename="src/layouts/SiteLayout.astro"
+---
+import { SignedIn, SignedOut, UserButton, SignInButton } from "@clerk/astro/components";
+---
+
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <link rel="icon" type="image/svg+xml" href="/favicon.svg" />
+    <meta name="generator" content={Astro.generator} />
+  </head>
+  <body>
+    <header>
+      <h1>{title}</h1>
+      <nav>
+        <SignedOut>
+         <SignInButton mode="modal" />
+        </SignedOut>
+        <SignedIn>
+          <UserButton />
+        </SignedIn>
+      </nav>
+    </header>
+    <article>
+      <slot />
+    </article>
```

Comment:
Removing extra markup
```suggestion
```

Reply:
Even if we drop the `src/pages/index.astro`, this should remain. This is reasonable expectation for a layout file and its likely to cause confusion if its removed. Devs who are newer to Astro amy miss that it was removed and fail to add it back until.